File: kotlin-workshop/part-1-powermock/src/main/kotlin/service/ShippingService.kt
package service

import domain.Order
import domain.ShippingConfirmation
import domain.ShippingItem
import domain.ShippingRequest
import infrastructure.ShippingApi

/**
 * Shipping service that creates its own ShippingApi instance (new)
 * This forces us to use mockkConstructor to test it in isolation
 */
class ShippingService {
    
    fun createShipment(order: Order): ShippingConfirmation {
        val api = ShippingApi() // Hardcoded instantiation - bad practice!
        
        val request = ShippingRequest(
            orderId = order.id,
            customerId = order.customerId,
            items = order.items.map { ShippingItem(it.productId, it.quantity) }
        )
        
        return api.createShipment(request)
    }
    
    fun cancelShipment(trackingNumber: String): Boolean {
        val api = ShippingApi() // Again, creating new instance
        return api.cancelShipment(trackingNumber)
    }
}

File: typescript-workshop/part-1-module-mocks/src/service/ShippingService.ts
import { Order } from '../domain/Order';
import { ShippingConfirmation } from '../domain/ShippingConfirmation';

export class ShippingService {
  createShipment(order: Order): ShippingConfirmation {
    throw new Error('Not implemented — requires shipping API connection');
  }

  cancelShipment(trackingNumber: string): boolean {
    throw new Error('Not implemented — requires shipping API connection');
  }
}
